tests/unit/references/test_references_processors.py

```diff
@@ -0,0 +1,643 @@
+                'authors': [
+                    {
+                        'full_name': "O'Brian, Dara",
+                        'inspire_role': 'editor',
+                    },
+                ],
+            },
+        },
+    ]
+    result = [builder.obj]
+
+    assert validate(result, subschema) is None
+    assert expected == result
+
+
+def test_set_pubnote():
+    schema = load_schema('hep')
+    subschema = schema['properties']['references']
+
+    builder = ReferenceBuilder()
+
+    builder.set_pubnote('Nucl.Phys.,B360,362')
+
+    expected = [
+        {
+            'reference': {
+                'publication_info': {
+                    'artid': '362',
+                    'journal_title': 'Nucl.Phys.',
+                    'journal_volume': 'B360',
+                    'page_start': '362',
+                },
+            },
+        },
+    ]
+    result = [builder.obj]
+
+    assert validate(result, subschema) is None
+    assert expected == result
+
+
+def test_set_pubnote_falls_back_to_raw_refs():
+    schema = load_schema('hep')
+    subschema = schema['properties']['references']
+
+    builder = ReferenceBuilder()
+
+    builder.set_pubnote('not-a-valid-pubnote')
+
+    expected = [
+        {
+            'raw_refs': [
+                {
+                    'schema': 'text',
+                    'source': 'reference_builder',
+                    'value': 'not-a-valid-pubnote',
+                },
+            ],
+        },
+    ]
+    result = [builder.obj]
+
+    assert validate(result, subschema) is None
+    assert expected == result
+
+
+def test_set_publisher():
+    schema = load_schema('hep')
+    subschema = schema['properties']['references']
+
+    builder = ReferenceBuilder()
+
+    builder.set_publisher('Elsevier')
+
+    expected = [
+        {
+            'reference': {
+                'imprint': {
+                    'publisher': 'Elsevier',
+                },
+            },
+        },
+    ]
+    result = [builder.obj]
+
+    assert validate(result, subschema) is None
+    assert expected == result
+
+
+def test_add_report_number_handles_arxiv_ids():
+    schema = load_schema('hep')
+    subschema = schema['properties']['references']
+
+    builder = ReferenceBuilder()
+
+    builder.add_report_number('hep-th/0603001')
+
+    expected = [
+        {
+            'reference': {
+                'arxiv_eprint': 'hep-th/0603001',
+            },
+        },
+    ]
+    result = [builder.obj]
+
+    assert validate(result, subschema) is None
+    assert expected == result
+
+
+def test_add_report_number_falls_back_to_parent_report_numbers():
+    schema = load_schema('hep')
+    subschema = schema['properties']['references']
+
+    builder = ReferenceBuilder()
+
+    builder.add_report_number('CERN-EP-2017-047')
+
+    expected = [
+        {
+            'reference': {
+                'publication_info': {
+                    'parent_report_number': 'CERN-EP-2017-047',
```

Comment:
why not [`report_number`](https://michamos.github.io/inspire-schemas/records/elements/reference.html#report-number)?